cmd/fs-v1.go

```diff
@@ -48,6 +48,20 @@ var fsTreeWalkIgnoredErrs = []error{
 	errVolumeNotFound,
 }
 
+// policy.json and notification.xml allowed only in .minio.sys
+func IsValidObjectNameFS(bucket, object string) bool {
+	var reservedObjectNames = map[string]bool{
+		"policy.json":      true,
+		"notification.xml": true,
+	}
+
```

Comment:
This logic is tricky to understand. The situation of `bucket != minioMetaBucket`  and matching the reserved object names should be together i feel.